### PR TITLE
add very simple web ui to bootstrap server

### PR DESCRIPTION
**Pull request by sbernard31**
This is a very minimalist UI for our leshan-bs-server.

### CODE REVIEW

File: leshan-bs-server/src/main/resources/webapp/index.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        <script src="tag/bootstrap.tag" type="riot/tag"></script>
+        <script src="tag/bootstrap-modal.tag" type="riot/tag"></script>
+        
+        <script src="js/bsconfigstore.js"></script>
+
+        <script src="vendor/js/jquery-2.1.1.min.js"></script>
+        <script src="vendor/js/bootstrap.min.js"></script>
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/riot/2.3.11/riot+compiler.min.js"></script>
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/riot/2.3.11/riot+compiler.min.js"></script>
```
Comment (msangoi) [inline leshan-bs-server/src/main/resources/webapp/index.html]: twice? :)

File: leshan-bs-server/src/main/resources/webapp/index.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            <p class="text-muted text-center"><small>
+                Powered by: <a href="http://eclipse.org/leshan">Lightweight M2M Leshan Project</a>
+                - <a href="https://github.com/eclipse/leshan/issues/">Report a bug</a>
+                - Version : <a href="https://github.com/eclipse/leshan/tree/@commit-id@">@version@</a>
+            </small></p>
+          </div>
+        </div>
+    
+    <script>
+        bsConfigStore = new BsConfigStore();
```
Comment (phtrivier) [inline leshan-bs-server/src/main/resources/webapp/index.html]: It is supposed to be JS or the Riot language ?

File: leshan-bs-server/src/main/resources/webapp/tag/bootstrap-modal.tag [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+    <script>
+        var self = this;
+        this.on('mount', function() {
+            $('#bootstrap-modal').modal('show');
+            this.update();
+        });
+
+        default_uri(){
+            if (secMode.value  == "no_sec")
```
Comment (phtrivier) [inline leshan-bs-server/src/main/resources/webapp/tag/bootstrap-modal.tag]: should be `===` for consistency

File: leshan-bs-server/src/main/resources/webapp/tag/bootstrap-modal.tag [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+                return "coap://leshan.eclipse.org:5683";
+            else
+                return "coaps://leshan.eclipse.org:5684";
+        }
+
+        has_error(){
+            var endpoint_has_error = (typeof endpoint.error === "undefined" || endpoint.error);
```
Comment (phtrivier) [inline leshan-bs-server/src/main/resources/webapp/tag/bootstrap-modal.tag]: `endpoint.hasError === undefined || endpoint.hasError` should be enough

File: leshan-bs-server/src/main/resources/webapp/tag/bootstrap-modal.tag [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+        submit(){
+            var id = [];
+            var key = [];
+            if (secMode.value === "psk"){
+                id = self.fromAscii(pskId.value);
+                key = self.fromHex(pskVal.value);
+                console.log(key)
+            }
+            var uri
+            if (!dmUrl.value || dmUrl.value.length == 0)
```
Comment (phtrivier) [inline leshan-bs-server/src/main/resources/webapp/tag/bootstrap-modal.tag]: Dangerous if without braces ;)

**Comment by sbernard31** [comment]
I rebase this one on master and fix the issue above.
I also add the riotjs file in the repo as CQ seems to be approved.
=> https://dev.eclipse.org/ipzilla/show_bug.cgi?id=10484
Could I merge this one ? 
Should I open a CQ for the riot 2.3.13 ? :stuck_out_tongue:

**Comment by sophokles73** [comment]
Do we really want to include all of the bootstrap, riot etc resources in the code base? Couldn't we simply reference them via their CDN URIs and let the user's browser download them when they start the UI?

**Comment by sbernard31** [comment]
The idea is to be able to launch or develop the bootstrap server even if we have no internet access.
(There're still some place where internet  is not easily accessible like EclipseCon :p )

**Comment by sophokles73** [comment]
I see :-)

**Comment by jvermillard** [comment]
see https://github.com/eclipse/leshan/issues/14

**Comment by sbernard31** [comment]
integrated in master since commit ff5ecfbd758964b71ee32ae9abcda3026450e696

**Pull request closed without merging**